src/core/unit.c

```diff
@@ -1742,6 +1745,23 @@ int unit_load(Unit *u) {
                 unit_invalidate_cgroup_members_masks(u);
         }
 
+        /* For top-level cgroups, if not set during loading, restrict allowed CPUs based on manager value */
+        if(u->manager->defaults.allowed_cpus.allocated > 0 && UNIT_HAS_CGROUP_CONTEXT(u)) {
+                cc = unit_get_cgroup_context(u);
+                if(!cc)
+                        goto fail;
+
+                if(cc->cpuset_cpus.allocated == 0) {
+                        r = unit_default_cgroup_path(u, &unit_cgroup_path);
+                        if (r < 0 || unit_cgroup_path == NULL)
```

Comment:
coding style, we rely on C's downgrade-to-bool feature to check for non-NULL-ness. i.e. drop the `== NULL`.